File: Contest/ABC/Contest_177/c.py
N = int(input())
A = list(map(int,input().split()))
mod = 1000000007
sum_A = 0
for i in range(N):
    A[i] = A[i] % mod
    sum_A += A[i]

ans = 0
for i in range(N-1):
    sum_A -= A[i]
    ans = (ans + (A[i] * sum_A) % mod) % mod

print(ans)

File: Contest/ABC/Contest_177/c.cpp
#include <bits/stdc++.h>
using namespace std;

#define rep(i, n) for (int i = 0; i < (int)(n); i++)

int main() { 
    long long N, a, mod = 1000000007, ans = 0, sum_all = 0;
    cin >> N;
    vector<long long> A(N);
    rep(i, N) {
        cin >> a;
        A[i] = a % mod;
        sum_all += A[i];
    }

    for(long long i = 0; i < N-1; ++i) {
        sum_all -= A[i];
        // cout << sum_all << " : " << A[i] << endl; 
        long long temp= (A[i] * sum_all) % mod;
        ans = (ans + temp) % mod;
    }
    cout << ans << endl;

    return 0;
}
